emission/net/ext_service/habitica/auto_tasks/level_challenge.py

```diff
@@ -0,0 +1,210 @@
+import emission.analysis.result.metrics.time_grouping as earmt
+
+# AUTOCHECK: {"mapper": "level_challenge", "args": {"bike_duration_scale": 60,
+# 													"bike_distance_scale": 1000,
+# 													"distance_to_cover": 100,
+# 													"trips_to_cover": 10,
+# 													"days_to_cover": 7}}
+
+def give_points(userid, task, curr_state)
+	
+	level_1_repetition = 1
+	trips_to_cover = task["args"]["trips_to_cover"]
+	distance_to_cover = task["args"]["distance_to_cover"]
+	days_to_cover = task["args"]["days_to_cover"]
+
+	# Level 1 - 10 min bike ride
+	if len(task.habitica_task.history) < level_1_repetition:
+
+		bike_scale = task["args"]["bike_duration_scale"]
+
+		#Assuming this is the 1st level, the start timestamp has from the day's midnight
+		timestamp_from_db = arrow.now().replace(hour=0, minute=0, second=0, microsecond=0).timestamp
+		timestamp_now = arrow.now().timestamp
+
+        #Get metrics
+        summary_ts = earmt.group_by_timestamp(userid, timestamp_from_db, timestamp_now, None, earmts.get_duration)
+        logging.debug("Metrics response: %s" % summary_ts)
+
+        if summary_ts[last_ts_processed] == None:
+        	new_state = curr_state
+        else: 
+	        for item in summary_ts["result"]
+	        	try:
+	        		bike_duration += item.BICYCLING
+	        		logging.debug("bike_duration += %s" % item.BICYCLING)
+	        	except AttributeError:
+	        		logging.debug("no bike")
+
+	        logging.debug("Finished with bike_duration == %s" % bike_duration)
+	        bike_duration_minute = bike_duration//bike_scale
+	        method_uri_level_challenge = "/api/v3/tasks/"+ task.task_id + "/score/up"
+		    
+		    #reward user by scoring + habits
+		    # more than 10 min bike is +1
+		    if bike_duration_minute >= time_to_cover:
+		    	res = proxy.habiticaProxy(user_id, 'POST', method_uri_level_challenge, None)
+		    	logging.debug("Request to score bike points %s" % res)
+
+		    #update the bike duration ts
+		    new_state = {'last_timestamp': summary_ts["last_ts_processed"]}
+
+	    logging.debug("Returning %s" % new_state)
+	    return new_state
+
+	# Level 2 - 10 trips on bike (1 score up for each trips)
+	if len(task.habitica_task.history) >= level_1_repetition and len(task.habitica_task.history) < trips_to_cover + level_1_repetition:
+
+		timestamp_from_db = curr_state['last_timestamp']
+		timestamp_now = arrow.now().timestamp
+
+		#Get metrics
+		summary_ts = earmt.group_by_timestamp(userid, timestamp_from_db, timestamp_now, None, earmts.get_count)
+	    logging.debug("Metrics response: %s" % summary_ts)
+
+	    for item in summary_ts["result"]
+	    	try:
+	            bike_trips += item.BICYCLING
+	            logging.debug("bike_trips += %s" % item.BICYCLING)
+	        except AttributeError:
+	            logging.debug("no bike")
+
+	    logging.debug("Finished with bike_trips == %s" % bike_trips)
+
+	    method_uri_level_challenge = "/api/v3/tasks/"+ task.task_id + "/score/up"
```

Comment:
How does the user know what they are supposed to be doing now for the challenge? Are you going to edit the task description? If so, where are the calls for that?